iterators.tex

```diff
@@ -6369,54 +6369,73 @@
 of a \tcode{Range} type that knows its size in constant time with the
 \tcode{size} function.
 
-\begin{codeblock}
-// \expos
-template <class T>
-concept bool @\xname{SizedRangeLike}@ =
-  Range<T>@\newtxt{()}@ &&
-  requires(@\newtxt{const }@T t) {
-    { size(t) } -> Integral;
-    requires Convertible@\newtxt{To}@<decltype(size(t)),
-                           DifferenceType<IteratorType<T>>>@\newtxt{()}@;
-  };
+\begin{itemdecl}
+@\oldtxt{// \expos}@
+@\oldtxt{template <class T>}@
+@\oldtxt{concept bool \xname{SizedRangeLike} =}@
+  @\oldtxt{Range<T> \&\&}@
+  @\oldtxt{requires(T t) \{}@
+    @\oldtxt{\{ size(t) \} -> Integral;}@
+    @\oldtxt{requires Convertible<decltype(size(t)),}@
+                         @\oldtxt{DifferenceType<IteratorType<T>{}>{}>;}@
+  @\oldtxt{\};}@
 
-@\newtxt{template <class R>}@
-@\newtxt{struct is_sized_range;}@
+@\newtxt{template <class T>}@
+@\newtxt{struct disable_sized_range;}@
 
 template <class T>
 concept bool SizedRange@\newtxt{() \{}\oldtxt{ =}@
-  @\newtxt{return}@ @\xname{SizedRangeLike}@<T> && is_sized_range<T>::value;
+  @\oldtxt{\xname{SizedRangeLike}<T> \&\& is_sized_range<T>::value;}@
+  @\newtxt{return !disable_sized_range<T>::value \&\&}@
+    @\newtxt{Range<T>() \&\&}@
+    @\newtxt{requires (const remove_reference_t<T>\& t) \{}@
+      @\newtxt{\{ size(t) \} -> Integral;}@
+      @\newtxt{\{ size(t) \} -> ConvertibleTo<DifferenceType<IteratorType<T\&>{}>{}>;}@
+    @\newtxt{\};}@
 @\newtxt{\}}@
-\end{codeblock}
+\end{itemdecl}
 
+\begin{itemdescr}
 \pnum
-\newtxt{Given an object \tcode{t} of type \tcode{T}, \tcode{SizedRange<T>()} is satisfied if and
-only if \tcode{size(t)} returns the number of elements in \tcode{t}.}
+\newtxt{Given an object \tcode{t} of type \tcode{T}, \tcode{SizedRange<T>()}
+is satisfied if and only if \tcode{size(t)} returns the number of elements in
+\tcode{t}.}
 
 \pnum
-For any type \tcode{T}, \tcode{is_sized_range<T>} derives from
-\tcode{true_type} if \tcode{T} \oldtxt{models}\newtxt{meets the syntactic requirements of}
-\tcode{\xname{SizedRangeLike}}, and \tcode{false_type} otherwise.
+For any type \tcode{T}, \tcode{\oldtxt{is}\newtxt{disable}_sized_range<T>}
+derives from \oldtxt{\tcode{true_type} if \tcode{T} models
+\tcode{\xname{SizedRangeLike}}, and} \tcode{false_type} \newtxt{if \tcode{T}
+is a non-reference cv-unqualified type, and from \\
+\tcode{disable_sized_range<remove_cv_t<remove_reference_t<T>{}>{}>}} otherwise.
```

Comment:
There's a difference between defining `struct disable_sized_range : disable_sized_range<__uncvref_t<T>>` and testing `!disable_sized_range<__uncvref_t<T>>::value`. With the later, the user can, if she really wants, override the default for particular qualified types. My gut tells me we want to give the users that control, but my gut has been wrong before.